src/nni_manager/core/test/nnimanager.test.ts

```diff
@@ -141,11 +184,106 @@ describe('Unit test for nnimanager', function () {
+        return nniManager.updateExperimentProfile(experimentProfile, 'TRIAL_CONCURRENCY').then(() => {
+            nniManager.getExperimentProfile().then((updateProfile) => {
+                expect(updateProfile.params.trialConcurrency).to.be.equal(3)
+                });
+        }).catch((error) => {
+           assert.fail(error);
+      })
+    })
+
+    it('test updateExperimentProfile MAX_EXEC_DURATION',  () => {
+        return nniManager.updateExperimentProfile(experimentProfile, 'MAX_EXEC_DURATION').then(() => {
+            nniManager.getExperimentProfile().then((updateProfile) => {
+                expect(updateProfile.params.maxExecDuration).to.be.equal(6);
+            });
+        }).catch((error) => {
+            assert.fail(error);
+        })
+    })
+
+    it('test updateExperimentProfile SEARCH_SPACE',  () => {
+        return nniManager.updateExperimentProfile(experimentProfile, 'SEARCH_SPACE').then(() => {
+            nniManager.getExperimentProfile().then((updateProfile) => {
+                expect(updateProfile.params.searchSpace).to.be.equal('{"y":2}');
+            });
+        }).catch((error) => {
+            assert.fail(error);
+        })
+    })
+
+    it('test updateExperimentProfile MAX_TRIAL_NUM',  () => {
+        return nniManager.updateExperimentProfile(experimentProfile, 'MAX_TRIAL_NUM').then(() => {
+            nniManager.getExperimentProfile().then((updateProfile) => {
+                expect(updateProfile.params.maxTrialNum).to.be.equal(2);
+            });
+        }).catch((error) => {
+            assert.fail(error);
+        })
+    })
+
+    it('test getStatus', () => {
+        assert.strictEqual(nniManager.getStatus().status,'RUNNING');
+    })
+
+    it('test getMetricData with trialJobId', () => {
+        //query a exist trialJobId
+        return nniManager.getMetricData('4321', 'CUSTOM').then((metricData) => {
+            expect(metricData.length).to.be.equal(1);
+            expect(metricData[0].trialJobId).to.be.equal('4321');
+            expect(metricData[0].parameterId).to.be.equal('param1');
+        }).catch((error) => {
+            assert.fail(error);
+        })
+    })
+
+    it('test getMetricData with invalid trialJobId', () => {
+        //query an invalid trialJobId
+        return nniManager.getMetricData('43210', 'CUSTOM').then((metricData) => {
+            assert.fail();
+        }).catch((error) => {
+        })
+    })
+
+    it('test getTrialJobStatistics', () => {
+        // get 3 trial jobs (init, addCustomizedTrialJob, cancelTrialJobByUser)
+        return nniManager.getTrialJobStatistics().then(function (trialJobStatistics) {
+            expect(trialJobStatistics.length).to.be.equal(2);
+            expect(trialJobStatistics[0].trialJobStatus).to.be.equal('WAITING');
+            expect(trialJobStatistics[0].trialJobNumber).to.be.equal(2);
+            expect(trialJobStatistics[1].trialJobStatus).to.be.equal('USER_CANCELED');
```

Comment:
Here we should not assume 'WAITING' is in trialJobStatistics[0],  please use another way to check.